x-pack/plugins/monitoring/server/alerts/large_shard_size_alert.ts

```diff
@@ -0,0 +1,216 @@
+          shardSize,
+          instanceId: `${clusterUuid}:${shardIndex}`,
+          itemLabel: shardIndex,
+        },
+        clusterUuid,
+        ccs,
+      };
+    });
+  }
+
+  protected getUiMessage(alertState: AlertState, item: AlertData): AlertMessage {
+    const { shardIndex, shardSize } = item.meta as IndexShardSizeUIMeta;
+    return {
+      text: i18n.translate('xpack.monitoring.alerts.shardSize.ui.firingMessage', {
+        defaultMessage: `The following index: #start_link{shardIndex}#end_link has a large shard size of: {shardSize}GB at #absolute`,
+        values: {
+          shardIndex,
+          shardSize,
+        },
+      }),
+      nextSteps: [
+        createLink(
+          i18n.translate('xpack.monitoring.alerts.shardSize.ui.nextSteps.investigateIndex', {
+            defaultMessage: '#start_linkInvestigate detailed index stats#end_link',
+          }),
+          `elasticsearch/indices/${shardIndex}/advanced`,
+          AlertMessageTokenType.Link
+        ),
+        createLink(
+          i18n.translate('xpack.monitoring.alerts.shardSize.ui.nextSteps.sizeYourShards', {
+            defaultMessage: '#start_linkHow to size your shards (Docs)#end_link',
+          }),
+          `{elasticWebsiteUrl}guide/en/elasticsearch/reference/current/size-your-shards.html`
+        ),
+        createLink(
+          i18n.translate('xpack.monitoring.alerts.shardSize.ui.nextSteps.shardSizingBlog', {
+            defaultMessage: '#start_linkShard sizing tips (Blog)#end_link',
+          }),
+          `{elasticWebsiteUrl}blog/how-many-shards-should-i-have-in-my-elasticsearch-cluster`
+        ),
+      ],
+      tokens: [
+        {
+          startToken: '#absolute',
+          type: AlertMessageTokenType.Time,
+          isAbsolute: true,
+          isRelative: false,
+          timestamp: alertState.ui.triggeredMS,
+        } as AlertMessageTimeToken,
+        {
+          startToken: '#start_link',
+          endToken: '#end_link',
+          type: AlertMessageTokenType.Link,
+          url: `elasticsearch/indices/${shardIndex}`,
+        } as AlertMessageLinkToken,
+      ],
+    };
+  }
+
+  protected filterAlertInstance(
+    alertInstance: RawAlertInstance,
+    filters: Array<CommonAlertFilter & { shardIndex: string }>
+  ) {
+    const alertInstanceStates = alertInstance.state?.alertStates as AlertState[];
+    const alertFilter = filters?.find((filter) => filter.shardIndex);
+    if (!filters || !filters.length || !alertInstanceStates?.length || !alertFilter?.shardIndex) {
+      return alertInstance;
+    }
+    const alertStates = alertInstanceStates.filter(
+      ({ meta }) => (meta as IndexShardSizeStats).shardIndex === alertFilter.shardIndex
+    );
+    return { state: { alertStates } };
+  }
+
+  protected executeActions(
+    instance: AlertInstance,
+    { alertStates }: AlertInstanceState,
+    item: AlertData | null,
+    cluster: AlertCluster
+  ) {
+    const shardIndices = alertStates
```

Comment:
We might need to condense this a little or something. This is what I'm seeing:
<img width="1941" alt="Screen Shot 2021-01-29 at 10 33 45 AM" src="https://user-images.githubusercontent.com/56682/106295070-fc377a80-621d-11eb-8b7b-a8aa069cc7c7.png">

Maybe we need to cap it at some limit and put like ", and more" or something. WDYT?